tools/fuzzer/ast.h

```diff
@@ -268,6 +269,29 @@ class TernaryExpr {
   std::unique_ptr<Expr> rhs_;
 };
 
+class BooleanConstant {
+ public:
+  enum class Value : bool { False, True };
+
+  static constexpr int PRECEDENCE = 0;
+
+  static constexpr Value True = Value::True;
+  static constexpr Value False = Value::False;
```

Comment:
Is this a shortcut to write `BooleanConstant::True` instead of `BooleanConstant::Value::True`? If so, I think it breaks the logic a bit. One would expect `BooleanConstant::True` to return a `BooleanConstant` that represents `True`, not the underlying `Value`.